cli.go

```diff
@@ -246,13 +279,19 @@ func (cli *CLI) Run(args []string) int {
 			// shutdown
 			log.Printf("[INFO] (cli) signal received to initiate graceful shutdown: %v", sig)
 			cancel()
-			select {
-			case <-exitCh:
-				log.Printf("[INFO] (cli) graceful shutdown")
-				return ExitCodeOK
-			case <-time.After(10 * time.Second):
-				log.Printf("[INFO] (cli) graceful shutdown timed out, exiting")
-				return ExitCodeInterrupt
+			counter := 0
+			for {
+				select {
+				case <-exitCh:
+					counter++
+					log.Printf("[INFO] (cli) graceful shutdown api / controller")
+					if counter >= exitBufLen {
+						return ExitCodeOK
+					}
+				case <-time.After(10 * time.Second):
```

Comment:
Reading the channel 2x is a good way make sure the API and the controller Runs have exited. However, I think the added for-loop now resets the 10 second timer. Is this intentional?

Reply:
Not intentional! I wondered if it was reseting but on testing it "looked fine" which I shouldn't have trusted. Let me fix this. Thanks for pointing this out!

Reply:
I updated this so that the timer would decrement on the loop so that it wouldn't reset to 10 seconds.

Reply:
Works for me, thanks for updating it